src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -591,3 +646,288 @@ RouterExecutorEnd(QueryDesc *queryDesc)
 	queryDesc->estate = NULL;
 	queryDesc->totaltime = NULL;
 }
+
+/*
+ * Walks each TargetEntry of the query, evaluates sub-expressions without Vars.
+ */
+void ExecuteFunctions(Query *query)
+{
+	CmdType commandType = query->commandType;
+	ListCell *targetEntryCell = NULL;
+	Node *modifiedNode = NULL;
+
+	/* DELETE has no targetList */
+	if (!(commandType == CMD_INSERT) && !(commandType == CMD_UPDATE))
+	{
+		return;
+	}
+
+	foreach(targetEntryCell, query->targetList)
```

Comment:
> Not sure why that'd be that interesting though. Let's get the basic feature in?

Foremost, to avoid users having to learn that function call only work in certain parts of certain queries. Second, because there are legitimate use-cases for it. If it's minor then it's much cheaper to do it now, otherwise we probably won't get around to it for a while.

The other suggestion I'd make is to call ExecuteFunctions from master_modify_multiple_shards, such that users can do:

    SELECT master_modify_multiple_shards('DELETE FROM events WHERE event_time < now() - interval ''1 month''');